back/engines/commercial/analytics/spec/acceptance/analytics_registrations_spec.rb

```diff
@@ -4,62 +4,119 @@
 require 'rspec_api_documentation/dsl'
 
 resource 'Analytics - FactRegistrations model' do
-  explanation 'Queries to summarise registrations.'
+  explanation 'Queries to summarise registrations and invitations to register.'
 
   before do
     header 'Content-Type', 'application/json'
     admin_header_token
   end
 
-  post 'web_api/v1/analytics' do
-    before_all do
-      # Create 3 users - 2 registered in Sept 2022 (admin & citizen) and 1 in Oct 2022 (citizen)
-      create(:admin, registration_completed_at: '2022-09-01 10:15:00')
-      create(:user, registration_completed_at: '2022-09-15 16:30:00')
-      create(:user, registration_completed_at: '2022-10-1 16:30:00')
+  before_all do
+    # Create associated date dimensions
+    create(:dimension_date, date: Date.new(2022, 8, 1))
+    create(:dimension_date, date: Date.new(2022, 8, 15))
+    create(:dimension_date, date: Date.new(2022, 9, 1))
+  end
 
-      # Create associated date dimensions
-      create(:dimension_date, date: Date.new(2022, 9, 1))
-      create(:dimension_date, date: Date.new(2022, 9, 15))
-      create(:dimension_date, date: Date.new(2022, 10, 1))
-    end
+  context 'fully complete registrations' do
+    post 'web_api/v1/analytics' do
+      before_all do
+        # Create 3 users - 2 registered in Sept 2022 (admin & citizen) and 1 in Oct 2022 (citizen)
+        create(:admin, registration_completed_at: '2022-08-01 10:15:00')
+        create(:user, registration_completed_at: '2022-08-15 16:30:00')
+        create(:user, registration_completed_at: '2022-09-1 16:30:00')
+      end
+
+      example 'group complete registrations by month' do
+        do_request({
+          query: {
+            fact: 'registration',
+            groups: 'dimension_date_registration.month',
+            aggregations: {
+              all: 'count'
+            }
+          }
+        })
+        assert_status 200
+        expect(response_data).to match_array([
+          { 'dimension_date_registration.month': '2022-08', count: 2 },
+          { 'dimension_date_registration.month': '2022-09', count: 1 }
+        ])
+      end
 
-    example 'group registrations by month' do
-      do_request({
-        query: {
-          fact: 'registration',
-          groups: 'dimension_date_registration.month',
-          aggregations: {
-            all: 'count'
+      example 'filter complete registrations between dates and return citizens only' do
+        do_request({
+          query: {
+            fact: 'registration',
+            filters: {
+              dimension_date_registration: {
+                date: { from: '2022-08-01', to: '2022-08-31' }
+              },
+              dimension_user: {
+                role: ['citizen', nil]
```

Comment:
Just for my own edification, what's the use of `nil` here?

Reply:
That's a cut and paste error. The nil will do nothing here as all registrations have a user - must have copied the original query from the visits tests where there are visits without users

Reply:
Will fix with a couple of changes arising from Fran's API refactor that he merged yesterday